CMakeLists.txt

```diff
@@ -1,17 +1,50 @@
+cmake_minimum_required(VERSION 3.10)
+project(ChunkTypesProject CXX)
 
-# Min CMake version
```

Comment:
Not entirely sure how the ESP dir gets used now

Reply:
it doesn’t right now, need to set my espidf back up and make a test project to see how i can import and use Chunk_Types

Reply:
now it does like so:
```cmake
# generic setup

set(SOURCE_DIR "source")
set(INCLUDE_DIR "include")
set(SOURCE_TESTS_DIR "source_tests")
set(INCLUDE_TESTS_DIR "include_tests”)

if(DEFINED ENV{ESP_PLATFORM})
    message(STATUS "Building for ESP32")

    idf_component_register(
        SRCS
        ...
    )
    
    # esp specific config

else()
    message(STATUS "Building for Unix")
    
    # unix specific setup

endif()
```

Reply:
still not a perfect automated system since it requires you to set a var before building for esp

Reply:
okay i’ve updated it to check for `IDF_PATH` rather which should only be present in the ESP IDF terminal or after specifically running the export function